<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="6" passed="5" failed="1" skipped="0">
  <reporter-output>
    <line>
      <![CDATA[demo4 class method]]>
    </line>
    <line>
      <![CDATA[demo5 class method]]>
    </line>
    <line>
      <![CDATA[demo1 class method]]>
    </line>
    <line>
      <![CDATA[demo2 class method]]>
    </line>
    <line>
      <![CDATA[demo3 class method]]>
    </line>
  </reporter-output>
  <suite started-at="2022-06-26T11:42:04 IST" name="Default suite" finished-at="2022-06-26T11:42:05 IST" duration-ms="540">
    <groups>
    </groups>
    <test started-at="2022-06-26T11:42:04 IST" name="Default test" finished-at="2022-06-26T11:42:05 IST" duration-ms="540">
      <class name="testNGPack.Demo">
        <test-method signature="a()[pri:0, instance:testNGPack.Demo@e7edb54]" started-at="2022-06-26T11:42:05 IST" name="a" description="my test case a is passed" finished-at="2022-06-26T11:42:05 IST" duration-ms="73" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: null
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.fail(Assert.java:106)
at testNGPack.Demo.a(Demo.java:11)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- a -->
        <test-method signature="e()[pri:0, instance:testNGPack.Demo@e7edb54]" started-at="2022-06-26T11:42:05 IST" name="e" finished-at="2022-06-26T11:42:05 IST" duration-ms="16" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[demo4 class method]]>
            </line>
          </reporter-output>
        </test-method> <!-- e -->
        <test-method signature="f()[pri:0, instance:testNGPack.Demo@e7edb54]" started-at="2022-06-26T11:42:05 IST" name="f" finished-at="2022-06-26T11:42:05 IST" duration-ms="9" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[demo5 class method]]>
            </line>
          </reporter-output>
        </test-method> <!-- f -->
        <test-method signature="b()[pri:0, instance:testNGPack.Demo@e7edb54]" started-at="2022-06-26T11:42:05 IST" name="b" finished-at="2022-06-26T11:42:05 IST" duration-ms="7" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[demo1 class method]]>
            </line>
          </reporter-output>
        </test-method> <!-- b -->
        <test-method signature="c()[pri:0, instance:testNGPack.Demo@e7edb54]" started-at="2022-06-26T11:42:05 IST" name="c" finished-at="2022-06-26T11:42:05 IST" duration-ms="3" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[demo2 class method]]>
            </line>
          </reporter-output>
        </test-method> <!-- c -->
        <test-method signature="d()[pri:0, instance:testNGPack.Demo@e7edb54]" started-at="2022-06-26T11:42:05 IST" name="d" finished-at="2022-06-26T11:42:05 IST" depends-on-methods="testNGPack.Demo.a" duration-ms="3" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[demo3 class method]]>
            </line>
          </reporter-output>
        </test-method> <!-- d -->
      </class> <!-- testNGPack.Demo -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
